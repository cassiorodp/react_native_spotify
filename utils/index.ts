import * as SecureStore from 'expo-secure-store';

const secureStore = async (key: string, value: string) => {
  await SecureStore.setItemAsync(key, value);
};

const getSecureStore = (key: string) => {
  return SecureStore.getItem(key);
};

export { secureStore, getSecureStore };
