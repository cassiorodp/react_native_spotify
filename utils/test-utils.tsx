import {
  RenderAPI,
  RenderOptions,
  render,
} from '@testing-library/react-native';
import { QueryClient, QueryClientProvider } from 'react-query';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  options?: RenderOptions
): RenderAPI => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
