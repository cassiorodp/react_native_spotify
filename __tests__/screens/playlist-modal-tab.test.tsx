import PlaylistModal from '@/app/tabs/playlists/playlist-modal';
import { render } from '@/utils/test-utils';

describe('Playlist modal tab component', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<PlaylistModal />);

    expect(toJSON()).toMatchSnapshot();
  });
});
