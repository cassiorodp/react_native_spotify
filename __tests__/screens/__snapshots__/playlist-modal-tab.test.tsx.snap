// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist modal tab component should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "backgroundColor": "rgb(255, 255, 255)",
      },
      {
        "flex": 1,
        "paddingHorizontal": 30,
      },
    ]
  }
>
  <View
    style={
      {
        "marginLeft": "auto",
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "gap": 22.5,
        "marginTop": 37.5,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "rgb(28, 28, 30)",
          },
          {
            "fontFamily": "Rubik-Regular",
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      DÃª um nome a sua playlist
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        [
          {
            "borderBottomWidth": 0.5,
            "fontFamily": "Rubik-Bold",
            "fontSize": 24,
            "textAlign": "center",
            "width": "80 %",
          },
          {
            "borderBottomColor": "rgb(216, 216, 216)",
            "color": "rgb(28, 28, 30)",
          },
        ]
      }
      testID="playlist-name-input"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "rgb(0, 122, 255)50",
          "borderRadius": 100,
          "marginTop": 10,
          "opacity": 1,
          "paddingHorizontal": 40,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "rgb(28, 28, 30)",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            [
              {
                "fontFamily": "Rubik-Bold",
                "fontSize": 16,
              },
              {
                "color": "rgb(242, 242, 242)",
              },
            ],
            undefined,
            undefined,
          ]
        }
      >
        Criar
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
